system/systemd.py

```diff
@@ -0,0 +1,329 @@
+    # init
+    module = AnsibleModule(
+        argument_spec = dict(
+                name = dict(required=True, type='str'),
+                state = dict(choices=[ 'started', 'stopped', 'restarted', 'reloaded'], type='str'),
+                enabled = dict(type='bool'),
+                masked = dict(type='bool'),
+                daemon_reload= dict(type='bool', default=False, aliases=['daemon-reload']),
+            ),
+            supports_check_mode=True,
+            required_one_of=[['state', 'enabled', 'masked', 'daemon_reload']],
+        )
+
+    # initialize
+    systemctl = module.get_bin_path('systemctl')
+    unit = module.params['name']
+    rc = 0
+    out = err = ''
+    result = {
+        'name':  unit,
+        'changed': False,
+        'status': {},
+    }
+
+    #TODO: check if service exists
+    (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
+    if rc != 0:
+        module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, unit, err))
+
+    # load return of systemctl show into dictionary for easy access and return
+    for line in out.split('\n'):
+        if line.strip() and '=' in line:
+            k,v = line.split('=', 1)
+            result['status'][k] = v.strip()
+
+    if 'LoadState' in result['status'] and result['status']['LoadState'] == 'not-found':
+        module.fail_json(msg='Could not find the requested service "%r": %s' % (unit, err))
+    elif 'LoadError' in result['status']:
+        module.fail_json(msg="Failed to get the service status '%s': %s" % (unit, result['status']['LoadError']))
+
+    # Run daemon-reload first, if requested
+    if module.params['daemon_reload']:
+        (rc, out, err) = module.run_command("%s daemon-reload" % (systemctl))
+        if rc != 0:
+            module.fail_json(msg='failure %d during daemon-reload: %s' % (rc, err))
+
+    # mask/unmask the service, if requested
+    if module.params['masked'] is not None:
+        masked = (result['status']['LoadState'] == 'masked')
+
+        # Change?
+        if masked != module.params['masked']:
+            result['changed'] = True
+            if module.params['masked']:
+                action = 'mask'
+            else:
+                action = 'unmask'
+
+            if not module.check_mode:
+                (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
+                if rc != 0:
+                    module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, err))
+
+    # Enable/disable service startup at boot if requested
+    if module.params['enabled'] is not None:
+        # do we need to enable the service?
+        enabled = False
+        (rc, out, err) = module.run_command("%s is-enabled '%s'" % (systemctl, unit))
+
+        # check systemctl result or if it is a init script
+        initscript = '/etc/init.d/' + unit
+        if rc == 0 or (os.access(initscript, os.X_OK) and bool(glob.glob('/etc/rc?.d/S??' + unit))):
```

Comment:
> it will fail to enable masked services

```sh
$ systemctl is-enabled ssh
masked

$ ls -l /etc/init.d/ssh
-rwxr-xr-x 1 root root 4077 Mar 21 12:06 /etc/init.d/ssh

$ find /etc/rc[0-9]* -name '*ssh*'
/etc/rc2.d/S02ssh
/etc/rc3.d/S02ssh
/etc/rc4.d/S02ssh
/etc/rc5.d/S02ssh
```

So, `if rc == 0 or (os.access(initscript, os.X_OK) and bool(glob.glob('/etc/rc?.d/S??' + unit))):` returns `True`. No?